### PR TITLE
Allow ArrayContainsKudf to search values on arrays inside a struct.

### LINKED ISSUE #1960: ARRAYCONTAINS with JSON Array generates error
KSQL-CLI 5.0.0 using a message with a JSON body. I cannot find a format for using ARRAYCONTAINS that returns as a result. I suspect that this issue is related to #1562. I notice that the array values after creating a stream no longer have surrounding "". 

`CREATE STREAM report_four (obj STRUCT<tags ARRAY<STRING>>) WITH (kafka_topic='create-json', value_format='JSON');`

```
ksql> DESCRIBE REPORT_FOUR;

Name                 : REPORT_FOUR
 Field   | Type                                
-----------------------------------------------
 ROWTIME | BIGINT           (system)           
 ROWKEY  | VARCHAR(STRING)  (system)           
 OBJ     | STRUCT<TAGS ARRAY<VARCHAR(STRING)>> 
-----------------------------------------------
For runtime statistics and query details run: DESCRIBE EXTENDED <Stream,Table>;
```

```
 SELECT OBJ->TAGS FROM report_four;
[microshare, location, battery]
[PTW, Test, 355, Tour Bus, public]
```
```
ksql> SELECT * FROM report_four WHERE ARRAYCONTAINS(OBJ->TAGS,'microshare');
Failed to generate code for filterExpression:ARRAYCONTAINS(FETCH_FIELD_FROM_STRUCT(REPORT_FOUR.OBJ, 'TAGS'), 'microshare') schema:org.apache.kafka.connect.data.SchemaBuilder@56c8e169
Caused by: Failed to create instance of kudfClass class io.confluent.ksql.function.udf.json.ArrayContainsKudf for function ARRAYCONTAINS
Caused by: Class io.confluent.ksql.function.KsqlFunction can not access a member of class io.confluent.ksql.function.udf.json.ArrayContainsKudf with modifiers ""
ksql> SELECT * FROM report_four WHERE ARRAYCONTAINS(OBJ->TAGS,"microshare");
Column microshare cannot be resolved.
ksql> SELECT * FROM report_four WHERE ARRAYCONTAINS(OBJ->TAGS,MICROSHARE);
Column MICROSHARE cannot be resolved.
ksql> SELECT ARRAYCONTAINS(OBJ->TAGS,'microshare') FROM report_four;
Unexpected error generating code for SelectValueMapper
Caused by: Failed to create instance of kudfClass class io.confluent.ksql.function.udf.json.ArrayContainsKudf for function ARRAYCONTAINS
Caused by: Class io.confluent.ksql.function.KsqlFunction can not access a member of class io.confluent.ksql.function.udf.json.ArrayContainsKudf with modifiers ""
```

Perhaps I am misusing ARRAYCONTAINS. There isn't much in the way of documentation or test cases for this scalar.

### PR DESCRIPTION
**Pull request by spena**
Fixes #1960

### Description 
This fixes an issue when using ARRAYCONTAINS() with an ARRAY inside a STRUCT.

For example, this below stream with a `STRUCT<colours ARRAY<VARCHAR>>` field.
```
ksql> CREATE STREAM NESTED_ARRAY_TEST (id BIGINT, first_name VARCHAR, last_name VARCHAR, email VARCHAR, preferred STRUCT<colours ARRAY<VARCHAR>>) WITH (kafka_topic='nested_array_test', value_format='JSON');
```

If a SELECT ... ARRAYCONTAINS(PREFERRED->COLOURS,'Pink') ... is used, then the following error happens:
```
ksql> SELECT FIRST_NAME, PREFERRED->COLOURS FROM NESTED_ARRAY_TEST WHERE ARRAYCONTAINS(PREFERRED->COLOURS,'Pink');
Failed to generate code for filterExpression:ARRAYCONTAINS(FETCH_FIELD_FROM_STRUCT(NESTED_ARRAY_TEST.PREFERRED, 'COLOURS'), 'Pink') schema:org.apache.kafka.connect.data.SchemaBuilder@76ea44ac
Caused by: Failed to create instance of kudfClass class io.confluent.ksql.function.udf.json.ArrayContainsKudf for function ARRAYCONTAINS
Caused by: Class io.confluent.ksql.function.KsqlFunction can not access a member of class io.confluent.ksql.function.udf.json.ArrayContainsKudf with modifiers ""
```

The error was happening because the ArrayContainsKudf() was not public, so the class couldn't be instantiated. Also, once fixed the above error, the problem was that ArrayContainsKudf.evaluate() needs to accept a List as a parameter as well.

### Testing done 
Added tests on ArrayContainsKudfTest.

### Reviewer checklist
- [ ] Ensure docs are updated if necessary. (eg. if a user visible feature is being added or changed).
- [ ] Ensure relevant issues are linked (description should include text like "Fixes #<issue number>")

### CODE REVIEW

**Comment by rodesai** [comment]
So is ARRAYCONTAINS just completely broken? I don't think we ever actually represent arrays as Java arrays - they are always Lists.

File: ksql-engine/src/main/java/io/confluent/ksql/function/udf/json/ArrayContainsKudf.java [hunk 1]
```diff
@@ -75,7 +76,10 @@ public Object evaluate(final Object... args) {
       return jsonStringArrayContains(searchValue, (String) args[0]);
     } else if (args[0] instanceof Object[]) {
       return ArrayUtil.containsValue(searchValue, (Object[]) args[0]);
+    } else if (args[0] instanceof Collection) {
```
Comment (rodesai) [inline ksql-engine/src/main/java/io/confluent/ksql/function/udf/json/ArrayContainsKudf.java]: This should always be either a List or a String

Comment (spena) [reply ksql-engine/src/main/java/io/confluent/ksql/function/udf/json/ArrayContainsKudf.java]: I removed the Object[] condition in a new patch. I verified that JSON and AVRO formats can still use ArrayContains without issues.

Comment (spena) [reply ksql-engine/src/main/java/io/confluent/ksql/function/udf/json/ArrayContainsKudf.java]: Btw, I added tests on QueryTranslationTest to verify ARRAYCONTAINS works as expected and it is not broken again.

**Pull request merged**